### PR TITLE
Improve concept page for Ingress

### PR DESCRIPTION
**Pull request by sftim**
- [Reword](https://github.com/kubernetes/website/pull/14313/commits/ac27b6ddd23acc71462721ca8d56e70585f24d3d) and reformat some definitions under [Terminology](https://kubernetes.io/docs/concepts/services-networking/ingress/#terminology)
- Highlight that Ingress is in beta
- Clarify the prerequistives. 
  - Drop details about GKE Ingress being beta, as there are lots of other ingress controllers (and not enough room to list shortcomings for all of them)
- Formatting fixes
- Write “load balancer” as two words
- Write “SSL / TLS” instead of “SSL”
- Use glossary tooltips where appropriate

Helps with #4478

### CODE REVIEW

**Comment by sftim** [comment]
Comments & feedback welcome!

File: content/en/docs/concepts/services-networking/ingress.md [hunk 1]
```diff
@@ -7,23 +7,34 @@ weight: 40
...
 
 For the sake of clarity, this guide defines the following terms:
 
-* Node: A single virtual or physical machine in a Kubernetes cluster.
-* Cluster: A group of nodes firewalled from the internet, that are the primary compute resources managed by Kubernetes.
-* Edge router: A router that enforces the firewall policy for your cluster. This could be a gateway managed by a cloud provider or a physical piece of hardware.
-* Cluster network: A set of links, logical or physical, that facilitate communication within a cluster according to the [Kubernetes networking model](/docs/concepts/cluster-administration/networking/).
-* Service: A Kubernetes [Service](/docs/concepts/services-networking/service/) that identifies a set of pods using label selectors. Unless mentioned otherwise, Services are assumed to have virtual IPs only routable within the cluster network.
+Node
+: A worker machine in Kubernetes, part of a cluster.
```
Comment (DanyC97) [inline content/en/docs/concepts/services-networking/ingress.md]: nit: i guess we can also add `wotker/node/compute` as different folks are used with one of another.

File: content/en/docs/concepts/services-networking/ingress.md [hunk 2]
```diff
@@ -7,23 +7,34 @@ weight: 40
...
+
+Cluster
+: A set of Nodes that run containerized applications managed by Kubernetes. For this example, and in most common Kubernetes deployments, nodes in the cluster are not part of the public internet.
```
Comment (DanyC97) [inline content/en/docs/concepts/services-networking/ingress.md]: nit: a Cluster is a K8s control plane (aka masters) + workers/nodes/compute. Maybe there will be some value to add this info in the paragraph

**Comment by DanyC97** [comment]
couple of nits if you want to consider them
/lgtm

File: content/en/docs/concepts/services-networking/ingress.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/concepts/services-networking/ingress.md]: Maybe it'd be better to adopt the glossary definition for each term defined there. It doesn't seem a problem to use the generally accepted definition of a Kubernetes cluster, for example.

I'll think about this.

Comment (DanyC97) [reply content/en/docs/concepts/services-networking/ingress.md]: @sftim ignore my comment for now as there is a wider discussion on K8s mailer which involves sig-cluster-lifecycle/ architecture to address/ agree on the definition - so far no agreement

File: content/en/docs/concepts/services-networking/ingress.md [hunk 3]
```diff
@@ -35,33 +46,20 @@ Traffic routing is controlled by rules defined on the Ingress resource.
...
+An Ingress can be configured to give services externally-reachable URLs, load balance traffic, terminate SSL / TLS, and offer name based virtual hosting. An [Ingress controller](/docs/concepts/services-networking/ingress-controllers) is responsible for fulfilling the Ingress, usually with a load balancer, though it may also configure your edge router or additional frontends to help handle the traffic.
 
 An Ingress does not expose arbitrary ports or protocols. Exposing services other than HTTP and HTTPS to the internet typically
 uses a service of type [Service.Type=NodePort](/docs/concepts/services-networking/service/#nodeport) or
 [Service.Type=LoadBalancer](/docs/concepts/services-networking/service/#loadbalancer).
 
 ## Prerequisites
 
-{{< feature-state for_k8s_version="v1.1" state="beta" >}}
+You must have an [ingress controller](/docs/concepts/services-networking/ingress-controllers) to satisfy an Ingress. Only creating an Ingress resource has no effect.
```
Comment (cody-clark) [inline content/en/docs/concepts/services-networking/ingress.md]: I think we should be consistent on capitalizing Ingress or not. It looks like both happens on this page. Which do you think would be better, @sftim?

Comment (sftim) [reply content/en/docs/concepts/services-networking/ingress.md]: `kubectl get "ingress controller"` will fail: there's no `IngressController` resource defined for cluster operators to manage, and that's what I'm trying to convey here.

(vs. Ingress which you certainly can manage via `kubectl`). Does that make sense?

If there are other places where the capitalization doesn't make sense, I'm happy to reconsider / revise.

**Pull request merged**